### DISCUSSION TITLE
[remark-lint] How to make the table cells untouched.

### DISCUSSION THREAD

**Discussion by Ir1d** [remarkjs/remark#597] [Q&A] [answered]
Hi, I was trying to use remark-lint for a markdown table, but the cells contained `|` so remark couldn't parse it correctly.

```
| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a||b` | | |
```

I understand that I could use `\|` instead in the cells, but in that case it will break my renderer (using mkdocs to render markdown).
I tried to disabe `remark-lint-table-pipe-alignment` and `remark-lint-table-cell-padding`, but they dont work. 
The lint results look like this:

```
|Pascal|C++|||||
|------|---|---|---|---|---|
| `not(a)` | `!a` |||||
| `a and b` | `a && b` |||||
| `a or b` |\`a||b\`|||
```

I wonder if there's a way to let the cells untouched? Thanks in advance.

My .remarkrc is here:

```
{
  "settings": {
    "listItemIndent": "mixed",
    "emphasis": "*",
    "spacedTable": false,
    "paddedTable": false
  },
  "plugins": [
    ["remark-lint-table-pipe-alignment", false],
    ["remark-lint-table-cell-padding", false]
  ]
}

```

My expected result: (at least the cells contain the right contents)

```
| Pascal      | C++        |     |     |
| ---------- | ---------- | --- | --- |
|  `not(a)`   |  `!a`      |     |     |
|  `a and b`  |  `a && b`  |     |     |
|  `a or b`   | `a||b` | | |
```

**Comment by wooorm** [comment, marked as answer]
the remark lint rules have to do with warnings, and as you’re not seeing warnings, those are apparently not the problem.

remark-stringify formats markdown, it passes options to a different project. This is documented here: https://github.com/remarkjs/remark/tree/main/packages/remark-stringify. As you can see `spacedTable` and `paddedTable` are not options there. They used to exist but have new names now: this change is documented in the changelog: https://github.com/remarkjs/remark/releases/tag/13.0.0.

remark adheres to standards. CommonMark by default, and if `remark-gfm` is used, tables and such (GFM features) are supported too.
But: `remark-gfm` does not support non-standard tables.

You could overwrite the serializer, by creating and maintaining your own version of `remark-gfm`, that parses and formats tables in the custom way you want. Or you could use a regex afterwards, to make GFM tables match what mkdocs expects?

**Reply by wooorm** [reply]
Or you could file the issue with `mkdocs`: they’re not following GFM. Maybe they should
